iquip/apps/builder.py

```diff
@@ -456,20 +460,33 @@ def initArgsEntry(self, experimentInfo: ExperimentInfo):
         Args:
             experimentInfo: The experiment information.
         """
+        nonScanEntry = ["BooleanValue", "StringValue", "EnumerationValue", "NumberValue"]
         for argName, (argInfo, *_) in experimentInfo.arginfo.items():
             # TODO(BECATRUE): The other types such as 'Scannable'
             # will be implemented in Basic Runner project.
-            entryCls = {
-                "BooleanValue": _BooleanEntry,
-                "StringValue": _StringEntry,
-                "EnumerationValue": _EnumerationEntry,
-                "NumberValue": _NumberEntry
-            }[argInfo.pop("ty")]
-            widget = entryCls(argName, argInfo)
-            item = QListWidgetItem(self.builderFrame.argsListWidget)
-            item.setSizeHint(widget.sizeHint())
-            self.builderFrame.argsListWidget.addItem(item)
-            self.builderFrame.argsListWidget.setItemWidget(item, widget)
+            argType = argInfo.pop("ty")
+            if argType in nonScanEntry:
+                entryCls = {
+                    "BooleanValue": _BooleanEntry,
+                    "StringValue": _StringEntry,
+                    "EnumerationValue": _EnumerationEntry,
+                    "NumberValue": _NumberEntry,
+                }[argType]
+                widget = entryCls(argName, argInfo)
+                item = QListWidgetItem(self.builderFrame.argsListWidget)
+                item.setSizeHint(widget.sizeHint())
+                self.builderFrame.argsListWidget.addItem(item)
+                self.builderFrame.argsListWidget.setItemWidget(item, widget)
+            elif argType == "Scannable":
+                widget = _ScanEntry(argName, argInfo)
+                item = QListWidgetItem(self.builderFrame.scanListWidget)
+                item.setSizeHint(widget.sizeHint())
+                self.builderFrame.scanListWidget.addItem(item)
+                self.builderFrame.scanListWidget.setItemWidget(item, widget)
```

Comment:
I think you can reduce the duplicated part.

For example,
```python
entryCls = {
    "BooleanValue": _BooleanEntry,
    "StringValue": _StringEntry,
    "EnumerationValue": _EnumerationEntry,
    "NumberValue": _NumberEntry,
    "Scannable": _ScanEntry,
}[argType]
widget = entryCls(argName, argInfo)
listWidget = self.builderFrame.scanListWidget if argType == "Scannable" else self.builderFrame.argsListWidget
item = QListWidgetItem(listWidget)
item.setSizeHint(widget.sizeHint())
listWidget.addItem(item)
listWidget.setItemWidget(item, widget)
```

or 

```python
entryCls, listWidget = {
    "BooleanValue": (_BooleanEntry, self.builderFrame.argsListWidget),
    "StringValue": (_StringEntry, self.builderFrame.argsListWidget),
    "EnumerationValue": (_EnumerationEntry, self.builderFrame.argsListWidget),
    "NumberValue": (_NumberEntry, self.builderFrame.argsListWidget),
    "Scannable": (_ScanEntry, self.builderFrame.scanListWidget),
}[argType]
widget = entryCls(argName, argInfo)
item = QListWidgetItem(listWidget)
item.setSizeHint(widget.sizeHint())
listWidget.addItem(item)
listWidget.setItemWidget(item, widget)
```

I think there are more neat method!

Reply:
That's a very good idea!
Thanks for the feedback.